esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2092,52 +2055,75 @@ void WsWuInfo::getWorkunitThorSlaveLog(const char *groupName, const char *ipAddr
     OwnedIFileIOStream ios = createBufferedIOStream(rIO);
     if (slaveNum > 0)
     {
-        StringBuffer line;
-        bool eof = false;
-        bool include = false;
+        readWorkunitLog(ios.get(), buf, outFile);
+    }
+    else
+    {//legacy wuid
+        appendIOStreamContent(buf, ios.get(), forDownload);
+    }
+}
 
-        VStringBuffer startwuid("Started wuid=%s", wuid.str());
-        VStringBuffer endwuid("Finished wuid=%s", wuid.str());
+void WsWuInfo::readWorkunitLog(OwnedIFileIOStream ios, MemoryBuffer& buf, const char* outFile)
```

Comment:
this is going 'copy' the parameter passed. In this case, it will link the parameter passed,
then when 'ios' goes out of scope it will release it.

Did you intend it to be like this?
I think there are -ve leaks where it is called.

Normally you'd expect the method to take a regular unowned object, i.e. IFileIOStream * in this case.
and for the ownership to be managed elsewhere.

Reply:
Good catch. I will fix it.